util/mqtt.c

```diff
@@ -38,7 +38,11 @@
  * is established every call.
  */
 
+#include <stdlib.h>
+#include <string.h>
+#define _GVM_MQTT_INTERNAL
```

Comment:
Why is this needed together with moving the typedef to the header?
I can compile and run the example without these changes. 
mqtt_connect() can stay static too.

Reply:
That was done before I switched back to the global setup and therefore was using mqtt_t outside of mqtt as void but wanted to define it once without having to cast based on the function definition. This is obsolete now and can be moved back to mqtt.c